src/main/java/org/jabref/logic/importer/fetcher/IEEE.java

```diff
@@ -236,4 +229,24 @@ public String getName() {
     public Optional<HelpFile> getHelpPage() {
         return Optional.of(HelpFile.FETCHER_IEEEXPLORE);
     }
+
+    @Override
+    public URL getComplexQueryURL(ComplexSearchQuery complexSearchQuery) {
+        try {
+            URIBuilder uriBuilder = new URIBuilder("https://ieeexploreapi.ieee.org/api/v1/search/articles");
+            uriBuilder.addParameter("apikey", API_KEY);
+            complexSearchQuery.getDefaultField().ifPresent(defaultField -> uriBuilder.addParameter("querytext", defaultField));
+            complexSearchQuery.getAuthor().ifPresent(author -> uriBuilder.addParameter("author", author));
+            complexSearchQuery.getTitle().ifPresent(articleTitle -> uriBuilder.addParameter("article_title", articleTitle));
+            complexSearchQuery.getJournal().ifPresent(journalTitle -> uriBuilder.addParameter("publication_title", journalTitle));
+            complexSearchQuery.getFromYear().map(String::valueOf).ifPresent(year -> uriBuilder.addParameter("start_year", year));
+            complexSearchQuery.getToYear().map(String::valueOf).ifPresent(year -> uriBuilder.addParameter("end_year", year));
+
+            URLDownload.bypassSSLVerification();
+            return uriBuilder.build().toURL();
+        } catch (URISyntaxException | MalformedURLException ex) {
+            LOGGER.error("Error creating URL.", ex);
+            throw new IllegalStateException("Error during creation of URL.", ex);
```

Comment:
Can you please have a look how the other `getUrl***` methods handle the exception handling. I think they are allowed to simply throw them and the performSearch method then converts it to a FetcherException.

Reply:
Oh, the comment was missing here. The assumption was that per definition, this exception will never raised. Neverheless +1 for consistency 😅

Reply:
I changed this to be consistent with the other getURL methods.